components/src/Checkbox/CheckboxGroup.js

```diff
@@ -9,39 +9,44 @@ import { Fieldset } from "../Form";
 
 const getCheckboxButtons = props => {
   const checkboxButtons = React.Children.map(props.children, checkbox => {
-    const { value, disabled, required, ...checkboxProps } = checkbox.props;
+    const { value, disabled, required, onChange, ...checkboxProps } = checkbox.props;
     return (
       <Checkbox
         {...checkboxProps}
         value={value}
         disabled={props.disabled || disabled}
+        error={!!(props.errorMessage || props.errorList)}
         required={props.required || required}
         name={props.name}
         defaultChecked={props.defaultValue ? props.defaultValue.includes(value) : undefined}
         checked={props.checkedValue ? props.checkedValue.includes(value) : undefined}
-        onChange={props.onChange}
+        onChange={props.onChange || onChange}
       />
     );
   });
   return checkboxButtons;
 };
 
-const BaseCheckboxGroup = ({ className, error, labelText, helpText, requirementText, ...props }) => (
-  <Fieldset role="group" className={className} hasHelpText={!!helpText}>
-    <Box mb="x1">
-      <legend>
-        {labelText}
-        {requirementText && <RequirementText>{requirementText}</RequirementText>}
-      </legend>
-      {helpText && <HelpText>{helpText}</HelpText>}
-    </Box>
-    {getCheckboxButtons(props)}
-    {error && <InlineValidation mt="x1" message={error} />}
-  </Fieldset>
-);
+const BaseCheckboxGroup = ({ className, errorMessage, errorList, labelText, helpText, requirementText, ...props }) => {
+  const otherProps = { ...props, errorMessage, errorList };
+  return (
+    <Fieldset role="group" className={className} hasHelpText={!!helpText}>
+      <Box mb="x1">
+        <legend>
+          {labelText}
+          {requirementText && <RequirementText>{requirementText}</RequirementText>}
+        </legend>
+        {helpText && <HelpText>{helpText}</HelpText>}
+      </Box>
+      {getCheckboxButtons(otherProps)}
+      <InlineValidation mt="x1" errorMessage={errorMessage} errorList={errorList} />
+    </Fieldset>
+  );
+};
 
 BaseCheckboxGroup.propTypes = {
-  error: PropTypes.string,
+  errorMessage: PropTypes.string,
+  errorList: PropTypes.oneOfType([PropTypes.string, PropTypes.arrayOf(PropTypes.string)]),
```

Comment:
Why support arrays and strings for `errorList`?

Reply:
Removed the support for the string case, I agree it made more sense to enforce it to always be an array